mkdocs.yml

```diff
@@ -0,0 +1,25 @@
+site_name: Crystal
+use_directory_urls: false
+docs_dir: .
+repo_url: https://github.com/crystal-lang/crystal
+edit_uri: https://github.com/crystal-lang/crystal-book/tree/master/
+
+theme:
+  name: material
+  palette:
+    primary: black
+    accent: blue
+  icon:
+    repo: fontawesome/brands/github
+
+plugins:
+  - search
+  - same-dir
+  - literate-nav:
+      nav_file: SUMMARY.md
+  - section-index
```

Comment:
Disclaimer:
The plugins that this uses are my own.

* https://github.com/oprypin/mkdocs-same-dir
  Without it, we'd have to move every single file into a *docs* subdirectory for mkdocs to work. 
* https://github.com/oprypin/mkdocs-literate-nav
  Without it, we'd have to re-write the nav (*SUMMARY.md*) in YAML.
* https://github.com/oprypin/mkdocs-section-index
  Without it, we wouldn't be able to keep sections clickable with their own associated index pages. ([read more](https://github.com/crystal-lang/crystal-book/issues/261#issuecomment-730028305))

None of these are a very big deal to get rid of if you ever feel the need to.

Reply:
I'd really prefer to keep it simple and try to avoid needing additional plugins for features that don't really make a difference one way or another.

* Moving all content into `docs/` even sounds like a good idea to keep it separate from all the configuration and stuff.
* I'd rather write navigation in a structured format like YAML than a custom markdown-based variant. SUMMARY.md is written in Markdown because that's what gitbook needs. When moving to a differen generator, we can switch to the default format which is also better.
* I don't think subsection index pages are that useful and as expressed in #261 the content structure might even be better without it. I'm fine with keeping it, but it's probably not gonna work with YAML, is it?

Reply:
In terms of minimizing the diff (even keeping GitBook working) this is better as is. Then there can be small PRs specifically doing those ugly changes.

Reply:
Oh, I see. As a temporary solution, that's all fine.

Reply:
I'm fine with these custom plugins.

Reply:
1. Removed in https://github.com/crystal-lang/crystal-book/pull/434 :+1: 
2. Sent a pull request to remove: https://github.com/crystal-lang/crystal-book/pull/440 :-1: 
3. Requires actual content changes, will not do :-1: